REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/certificate/controller/CertificateController.java

```diff
@@ -0,0 +1,109 @@
+package com.epam.esm.certificate.controller;
+
+import com.epam.esm.certificate.entity.Certificate;
+import com.epam.esm.certificate.exception.CertificateInvalidRequestException;
+import com.epam.esm.certificate.service.CertificateService;
+import com.epam.esm.certificate.service.CertificateServiceImpl;
+import com.epam.esm.errorhandle.validation.Validate;
+import com.epam.esm.hateoas.HateoasSupport;
+import org.springdoc.api.annotations.ParameterObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/v1/api/certificate")
+public class CertificateController {
+
+    private final CertificateService certificateService;
+    private final HateoasSupport hateoasSupport;
+    private final Validate validate;
```

Comment:
`Validate` name sounds wrong to me. I would prefer `Validator` at least. 
Probably `ValidationRules`, so you can call it like 
```java
validationRules.isIdValid(id);
// or
validationRules.checkId(id);
```

Reply:
Validate class renamed to Validator. His id( ) method renamed to isPositiveAndParsableId( )